File: app/src/test/java/org/mozilla/focus/persistence/TabModelTest.java
/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.persistence;

import junit.framework.Assert;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

@RunWith(RobolectricTestRunner.class)
public class TabModelTest {

    @Test
    public void testSanity() {
        Assert.assertFalse(new TabModel(null, null).isValid());
        Assert.assertFalse(new TabModel(null, null, "title", "").isValid());
        Assert.assertFalse(new TabModel(null, null, null, "url").isValid());
        Assert.assertFalse(new TabModel("id", null, null, null).isValid());

        Assert.assertTrue(new TabModel("id", null, null, "url").isValid());
        Assert.assertTrue(new TabModel("id", null, "title", "url").isValid());
    }
}

File: components/feature/tabs/src/test/java/org/mozilla/rocket/tabs/TabModelTest.kt
package org.mozilla.rocket.tabs

import junit.framework.Assert.assertEquals
import org.junit.Test
import org.robolectric.annotation.Config
import java.util.UUID

@Config(manifest = Config.NONE)
class TabModelTest {

    @Test
    fun testIsValid() {
        TabModel(UUID.randomUUID().toString(),
                "parent_id",
                "title",
                "https://mozilla.org").let { model ->
            assertEquals(true, model.isValid())
        }

        TabModel(UUID.randomUUID().toString(),
                "",
                "",
                "https://mozilla.org").let { model ->
            assertEquals(true, model.isValid())
        }

        TabModel(UUID.randomUUID().toString(),
                null,
                null,
                "https://mozilla.org").let { model ->
            assertEquals(true, model.isValid())
        }

        // no id, invalid
        TabModel("", "", "", "https://mozilla.org").let { model ->
            assertEquals(false, model.isValid())
        }

        TabModel("", "", "", "").let { model ->
            assertEquals(false, model.isValid())
        }

        // no url, invalid
        TabModel(UUID.randomUUID().toString(), "", "", "").let { model ->
            assertEquals(false, model.isValid())
        }

        TabModel(UUID.randomUUID().toString(), "", "", null).let { model ->
            assertEquals(false, model.isValid())
        }
    }
}
